{
  T result=null;
  try {
    result=this.next.execute(processCallback,object,messageProcessor,event);
    return result;
  }
 catch (  Exception e) {
    if (processCallback.getManagedExceptions() != null) {
      for (      Class<? extends Exception> exceptionClass : processCallback.getManagedExceptions()) {
        if (exceptionClass.isInstance(e)) {
          if (((OAuth2Adapter)object).getRefreshToken() != null) {
            if (logger.isDebugEnabled()) {
              logger.debug("A managed exception has been thrown. Attempting to refresh access token.");
            }
            try {
              object.refreshAccessToken();
            }
 catch (            Exception newException) {
              if (logger.isDebugEnabled()) {
                logger.debug("Another exception was thrown while attempting to refresh the access token. Throwing original exception back up",newException);
              }
              throw e;
            }
            result=this.next.execute(processCallback,object,messageProcessor,event);
            return result;
          }
        }
      }
    }
    throw e;
  }
}
